src/Xenial.Tasty.Cli/Commands/InteractiveCommand.cs

```diff
@@ -0,0 +1,230 @@
+                            return 0;
+                        }
+                        catch (SimpleExec.NonZeroExitCodeException e)
+                        {
+                            return e.ExitCode;
+                        }
+                    }
+                }
+                return 0;
+            }
+            catch (TaskCanceledException)
+            {
+                return 1;
+            }
+        }
+
+        static Task WaitForInput(IList<SerializableTastyCommand> commands, TastyServer tastyServer)
+            => Promise(async (resolve) =>
+            {
+                Func<Task> cancelKeyPress = () => Promise((resolve, reject) =>
+                {
+                    Console.CancelKeyPress += (_, e) =>
+                    {
+                        if (e.Cancel)
+                        {
+                            Console.WriteLine("Cancelling execution...");
+                            reject();
+                        }
+                    };
+                });
+
+                Func<Task> endTestPipelineSignaled = () => Promise((resolve) =>
+                {
+                    tastyServer.EndTestPipelineSignaled = () =>
+                    {
+                        Console.WriteLine("Pipeline ended.");
+                        resolve();
+                    };
+                });
+
+                Func<Task> testPipelineCompletedSignaled = () => Promise((resolve) =>
+                {
+                    tastyServer.TestPipelineCompletedSignaled = () =>
+                    {
+                        Console.WriteLine("Pipeline completed.");
+                        resolve();
+                    };
+                });
+
+                Func<Task<ConsoleKeyInfo>> readConsoleKey = () => Promise<ConsoleKeyInfo>(resolve =>
+                {
+                    _ = Task.Run(() =>
+                    {
+                        var info = Console.ReadKey(true);
+                        resolve(info);
+                    });
+                });
+
+                Action writeCommands = () =>
+                {
+                    Console.WriteLine("Interactive Mode");
+                    Console.WriteLine("================");
+
+                    foreach (var c in commands)
+                    {
+                        Console.WriteLine($"{c.Name} - {c.Description}" + (c.IsDefault ? " (default)" : string.Empty));
+                    }
+
+                    Console.WriteLine("c - Cancel");
+                    Console.WriteLine("================");
+                };
+
+                Func<ConsoleKeyInfo, SerializableTastyCommand?> findCommand = (info) =>
+                {
+                    var key = info.Key.ToString().ToLower();
+
+                    var command = info.Key == ConsoleKey.Enter
+                        ? commands.FirstOrDefault(c => c.IsDefault)
+                        : commands.FirstOrDefault(c => c.Name.ToLowerInvariant() == key);
```

Comment:
Perhaps remove the `key` variable and avoid the double ToLower by doing something like this instead:
`string.Equals(c.Name, info.Key.ToString(),StringComparison.InvariantCultureIgnoreCase)`

Reply:
StringComparison.OrdinalIgnoreCase is the right choice, to not use InvariantCultureIgnoreCase

Reply:
Cause the commands are provided by the client (and can be extended by the user) `OrdinalIgnoreCase` should be the right choice, if used for example with non latin languages.